### DISCUSSION TITLE
[Feature request] - Templates for column groups

### DISCUSSION THREAD

**Discussion by susanu** [Laravel-Backpack/community-forum#1117] [Ideas]
Hi,

I really like this datagrid template for tabler theme: https://preview.tabler.io/datagrid.html
Any way we can implement this into show operation?
We can have a group of columns with this template and the rest with the normal template (the current one).
The current template for show operation is kinda boring.

Let me know what you think.

**Comment by pxpm** [comment]
Hey @susanu thanks for the suggestion.

Maybe as a widget ? Creating a `datagrid` widget that you can include anywhere ? If we make the "datagrid values" a closure that we evaluate at "run time", we can access the `app('crud')->getEntry()` to get the current entry and extract the needed information ?

```php
Widget::add()->type('datagrid')->cols(3)->inputs(['field1','field2','field3','field4','field5','field6'])
// or if we need more control, accept an array of arrays
[
    [
        'name' => 'field1',
        'label' => 'Field 1,
    ],
    [
    // ...
    ],
]
```
Wouldn't that "quite" work ? 

I agree with you that's a little bit boring. We introduced "tabs" a few time ago (probably it was you that suggested it), it makes it a little better ... but still 😿 
We have no concept of "templates" at the moment, introducing a new "architecture" is not as simple as adding a new widget. 

What do you think ?

**Comment by susanu** [comment]
Yeah, looks good to me.
